pkg/sql/virtual_schema.go

```diff
@@ -71,6 +76,95 @@ type virtualSchemaTable struct {
 	delegate func(ctx context.Context, p *planner, db *DatabaseDescriptor) (planNode, error)
 }
 
+// virtualSchemaView represents a view within a virtualSchema
+type virtualSchemaView struct {
+	schema string
+
+	// parent represents the table that the virtualSchemaView is augmenting
+	parent string
+
+	// extraCols represents the resultColumns that the view appends to the parent's resultColumns
+	extraCols sqlbase.ResultColumns
+}
+
+func (t virtualSchemaTable) getSchema() string {
+	return t.schema
+}
+
+func (t virtualSchemaView) getSchema() string {
+	return t.schema
+}
+
+func (t virtualSchemaTable) initVirtualTableDesc(
+	ctx context.Context, st *cluster.Settings,
+) (sqlbase.TableDescriptor, error) {
+	stmt, err := parser.ParseOne(t.schema)
+	if err != nil {
+		return sqlbase.TableDescriptor{}, err
+	}
+
+	create := stmt.(*tree.CreateTable)
+	for _, def := range create.Defs {
+		if d, ok := def.(*tree.ColumnTableDef); ok && d.HasDefaultExpr() {
+			return sqlbase.TableDescriptor{},
+				errors.Errorf("virtual tables are not allowed to use default exprs "+
+					"because bootstrapping: %s:%s", create.Table, d.Name)
+		}
+	}
+
+	// Virtual tables never use SERIAL so we need not process SERIAL
+	// types here.
+	return MakeTableDesc(
+		ctx,
+		nil, /* txn */
+		nil, /* vt */
+		st,
+		create,
+		0, /* parentID */
+		keys.VirtualDescriptorID,
+		hlc.Timestamp{}, /* creationTime */
+		publicSelectPrivileges,
+		nil, /* affected */
+		nil, /* semaCtx */
+		nil, /* evalCtx */
+	)
+}
+
+func (v virtualSchemaView) initVirtualTableDesc(
+	ctx context.Context, st *cluster.Settings, parentDesc *sqlbase.TableDescriptor,
+) (sqlbase.TableDescriptor, error) {
+	stmt, err := parser.ParseOne(v.schema)
+	if err != nil {
+		return sqlbase.TableDescriptor{}, err
+	}
+
+	create := stmt.(*tree.CreateView)
+
+	return MakeViewTableDesc(
+		create,
+		v.getResultColumns(parentDesc),
+		keys.DescriptorTableID,
```

Comment:
For some reason, using `0` as `parentID` and `keys.VirtualDescriptorID` as `ID` causes logictest to fail on the query `SELECT * FROM crdb_internal.ranges` (but this query works when using `./cockroach sql`).